packages/message-db/message-db-consumer/src/lib/MessageDbSource.mts

```diff
@@ -21,8 +21,12 @@ interface CreateOptions {
   handler: (streamName: StreamName, events: ITimelineEvent[]) => Promise<void>
   /** sleep time in ms between reads when at the end of the category */
   tailSleepIntervalMs: number
+  /** sleep time in ms between checkpoint commits */
```

Comment:
to avoid having two timespans beside each other, maybe put generic stuff like tailsleep and maxreadahead together

then checkpoint/group stuff

then dispatch/scheduling stuff such as maxConcurrentStreams

L26 is it still p-limit

Reply:
I'll take a look at a better config structure after merging this 👍